sources/Map/Map.cpp

```diff
@@ -3,30 +3,26 @@
 // Edited by Benjamin Henry on 5/26/21.
 //
 
-#include <Component/Collision/CollisionComponent.hpp>
+#include "Component/Collision/CollisionComponent.hpp"
+#include "System/Collision/CollisionSystem.hpp"
 #include "Map.hpp"
 #include <iostream>
 
 namespace RAY3D = RAY::Drawables::Drawables3D;
 
 namespace BBM
 {
-	void MapGenerator::wallCollide(WAL::Entity &entity, const WAL::Entity &wall)
+	void MapGenerator::wallCollide(WAL::Entity &entity, const WAL::Entity &wall, int collidedAxis)
 	{
 		auto *mov = entity.tryGetComponent<MovableComponent>();
+
 		if (!mov)
 			return;
-		auto &pos = entity.getComponent<PositionComponent>();
-		const auto &wallPos = wall.getComponent<PositionComponent>();
-		auto diff = pos.position + mov->getVelocity() - wallPos.position;
-//		mov->_velocity = Vector3f();
-		if (diff.x <= 0 && mov->_velocity.x < 0)
+		if (collidedAxis & CollisionComponent::CollidedAxis::X)
```

Comment:
You should check that you are indeed going in the direction of the wall, if you try to run away from a wall you will be disappointed if the move is canceled.

Reply:
wontfix